templates/default/sqlserver.yaml.erb

```diff
@@ -1,39 +1,9 @@
-init_config:
-    #
-    # By default, we only capture *some* of the metrics available in the
-    # `sys.dm_os_performance_counters` table. You can easily add additional
-    # metrics by following the custom_metrics structure shown below.
-    #
-    #custom_metrics:
-    #
-    #    This is a basic custom metric. There is not instance assicated with
-    #    this counter.
-    #    Options for type are: gauge, rate, histogram
-    #
-    #    -   name: sqlserver.clr.execution
-    #        type: gauge
-    #        counter_name: CLR Execution
-    #
-    #    This counter has multiple instances assocaited with it and we're
-    #    choosing to only fetch the '_Total' instance.
-    #
-    #    -   name: sqlserver.exec.in_progress
-    #        type: gauge
-    #        counter_name: OLEDB calls
-    #        instance_name: Cumulative execution time (ms) per second
-    #
-    #    This counter has multiple instances assocaited with it and we want
-    #    every instance available. We'll use the special case ALL instance
-    #    which *requires* a value for "tag_by". In this case, we'll get metrics
-    #    tagged as "db:mydb1", "db:mydb2".
-    #
-    #    -   name: sqlserver.db.commit_table_entries
-    #        type: gauge
-    #        counter_name: Log Flushes/sec
-    #        instance_name: ALL
-    #        tag_by: db
+# Generated by Chef, local modifications will be overwritten
 
-instances:
-    -   host: HOST,PORT
-        username: my_username
-        password: my_password
+<% if @init_config.empty? -%>
```

Comment:
Thanks @mlcooper for the explanation.

Since the behavior of having `init_config` set to the empty map (`{}`) instead of `null` is not a problem per se, what do you think of removing the `if`  clause from the template and setting the first line of the expected yaml in the spec test to `init_config: {}`?